### ISSUE TITLE
Datenanreicherung - Paketname

### ISSUE THREAD

**Issue by MiriKon** [hbz/laser-ygor#278]
Beim Eintragen des "Titels" (s. Screenshot) im Anreicherungsprozess mit Ygor sollten wir mal überlegen, dass etwas klarer wird, dass hier der Paketname definiert wird. Das könnte man über ein Fragezeichen machen. Für bereits bestehende Pakete, die man nur aktualisieren will, sollte man aber am allerbesten die Paket ID des schon bestehenden Pakets eingeben können, da dann das Paket eindeutiger identifiziert werden kann. Der gesamte Prozess sollte aber generell nochmals besprochen und genauer geplant werden.

![datenanreicherung_paketname_ygor](https://user-images.githubusercontent.com/45172978/91311153-08bd3000-e7b3-11ea-87ca-c0fe65848f65.jpg)

**Comment by philboeselager** [comment]
> Beim Eintragen des "Titels" (s. Screenshot) im Anreicherungsprozess mit Ygor sollten wir mal überlegen, dass etwas klarer wird, dass hier der Paketname definiert wird.

Statt "Titel" könnte dort auch "Pakettitel" oder "Paketname" stehen. @MiriKon Bitte sag gerne, wie es wohl am besten wäre.

> Für bereits bestehende Pakete, die man nur aktualisieren will, sollte man aber am allerbesten die Paket ID des schon bestehenden Pakets eingeben können, da dann das Paket eindeutiger identifiziert werden kann.

Jein. "Eindeutiger" nicht, da Paketnamen in der GOKb immer eindeutig sind. Es gibt keine 2x vorkommenden Paketnamen.
Die ID würde die Eingabe wohl dennoch vereinfachen. Das setzt allerdings voraus, dass die ID bekannt ist. @MiriKon Von welcher ID sprechen wir?

- die gesamte GOKb-URL (https://gokb.org/gokb/public/packageContent/org.gokb.cred.Package%3A4411609)
- UUID (8915f207-e95a-4301-9860-11d5c0182050 im o. g. Beispiel)
- Datenbank-ID (4411609)

**Comment by MiriKon** [comment]
"Paketname" wäre dann von mir bevorzugt und was die ID anbelangt, hatte ich an die Package ID gedacht, also z.B. für das Paket Emerald Management 120 würde ich dann diese ID 1224629 nehmen. Die ist dann ja auch die eindeutige ID nur für dieses Paket, korrekt? 

D.h. ich schlage folgendes vor: es heißt "Paketname" und danach kommt ein Fragezeichen-Icon:

"Für das Aktualisieren bereits in der GOKb existierender Pakete, können Sie hier auch die Paket ID angeben."

**Comment by philboeselager** [comment]
> Die ist dann ja auch die eindeutige ID nur für dieses Paket, korrekt?

Ja. Nur ist diese ID auf jedem Server unterschiedlich. Das heißt, sollte es unpraktisch sein, dass sich die ID auf Test- und Produktionsinstanz unterscheidet, wäre möglicherweise die UUID zu bevorzugen - welche allerdings unleserlicher ist.

**Comment by darupp** [comment]
Aus den von Philipp genannten Gründen darf man auf keinen Fall die Datenbank-ID nehmen, sondern nur die UUID, die proprietäre oder ggf. das Sigel gesucht werden kann. Die Datenbank-ID wird in der neuen Oberfläche auch nicht mehr angezeigt. 

Eine sinnvolle Lösung kann eine kombiniertes Texteingabe/Such-Dropdown-Feld sein, bei dem sowohl über Name wie über die genannten Identifier gesucht werden kann. Die Werte kommen dann, wie unten bei Provider und Anbieter, aus dem ES-Index. Wenn ein nicht in der Suchliste nicht vorhandener Text eingegeben wird, also ein neuer Paketname, muss das Oberflächenelement im Gegensatz zu klassischen Suchfeldern die Oberfläche diesen akzeptieren. Ich weiß nicht, ob dieses kombinierte Eingabelement schon out of the box im UI-Framework enthalten ist.

Diese Lösung ist auch konsistent mit dem neuen Erfassungsclient (siehe z.B. S. 35 zur Providersuche)
